### PR TITLE
Feature trial subscription

### PR DESCRIPTION
**Pull request by msmoradi**

### CODE REVIEW

File: poolakey/src/main/java/ir/cafebazaar/poolakey/billing/FeatureConfig.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package ir.cafebazaar.poolakey.billing
+
+import android.os.Bundle
+
+object FeatureConfig {
```
Comment (shayanpourvatan) [inline poolakey/src/main/java/ir/cafebazaar/poolakey/billing/FeatureConfig.kt:5]: Make it internal

File: poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt [hunk 2]
```diff
@@ -202,6 +216,62 @@ internal class ReceiverBillingConnection(
         }.run(::sendBroadcast)
     }
 
+    private fun getFeatureConfig(
+        callback: GetFeatureConfigCallback.() -> Unit
+    ) {
+        featureConfigCallback = callback
```
Comment (shayanpourvatan) [inline poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt:222]: You need version check here

File: poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt [hunk 3]
```diff
@@ -411,7 +518,9 @@ internal class ReceiverBillingConnection(
     }
 
     companion object {
+
         private const val BAZAAR_WITH_RECEIVER_CONNECTION_VERSION = 801301
+        private const val BAZAAR_WITH_FEATURE_CONFIG_VERSION = 801301
```
Comment (shayanpourvatan) [inline poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt:523]: Change this version code based on released version

File: poolakey/src/main/java/ir/cafebazaar/poolakey/callback/GetFeatureConfigCallback.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package ir.cafebazaar.poolakey.callback
+
+import android.os.Bundle
+import java.lang.Exception
+
+class GetFeatureConfigCallback {
```
Comment (shayanpourvatan) [inline poolakey/src/main/java/ir/cafebazaar/poolakey/callback/GetFeatureConfigCallback.kt:6]: I think this must be internal as we check this in our side

File: poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt (hunk 2 above)
Comment (msmoradi) [reply poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt:222]: this function just sendBroadcast for getting the feature config , we check the version before it calls.

File: poolakey/src/main/java/ir/cafebazaar/poolakey/callback/GetFeatureConfigCallback.kt (hunk 4 above)
Comment (msmoradi) [reply poolakey/src/main/java/ir/cafebazaar/poolakey/callback/GetFeatureConfigCallback.kt]: yes , i think it would be better internal too.

**Comment by shayanpourvatan** [comment]
LGTM

**Review by hamidrezasahraei — APPROVED** [review]
LGTM

**Review by hamidrezasahraei — COMMENTED** [review]
LGTM

**Pull request merged**